microcosm/decorators.py

```diff
@@ -42,3 +42,24 @@ def get_defaults(func):
 
     """
     return getattr(func, DEFAULTS, {})
+
+
+def iter_keys(dct, prefix=""):
+    for key, value in dct.items():
+        if isinstance(value, dict):
+            yield from iter_keys(value, f"{key}.")
+        else:
+            yield f"{prefix}{key}"
+
+
+def public(loader):
+    """
+    Flag a loader as non-secret.
```

Comment:
This is a little misleading. Looks like we're really flagging all the keys that come from a specific loader as public. We could, in theory, update the definition of the loader itself to have some sort of "public" flag to treat all data coming from this loader as public.

I think this implementation makes sense, so maybe it's just worth emphasizing we're making the keys that come from a loader public over an actual property on the loader?

Reply:
I thought about this a bit more on the way home. I think I'm going to scrap this implementation in favor of having more control over the configuration loading mechanism used in `create_object_graph` and write a loader that partitions values into secret and non-secret categories.

This makes it easier to trace specific values (and not just keys) into one side of the partition, in case a secret overrides a non-secret.